x-pack/plugins/enterprise_search/public/applications/shared/role_mapping/users_table.tsx

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React from 'react';
+
+import { upperFirst } from 'lodash';
+
+import { EuiBadge, EuiBasicTableColumn, EuiInMemoryTable, EuiTextColor } from '@elastic/eui';
+
+import { ASRoleMapping } from '../../app_search/types';
+import { SingleUserRoleMapping } from '../../shared/types';
+import { WSRoleMapping } from '../../workplace_search/types';
+
+import {
+  INVITATION_PENDING_LABEL,
+  ALL_LABEL,
+  FILTER_USERS_LABEL,
+  NO_USERS_LABEL,
+} from './constants';
+
+import { UsersAndRolesRowActions } from './';
+
+interface AccessItem {
+  name: string;
+}
+
+interface SharedUser extends SingleUserRoleMapping<ASRoleMapping | WSRoleMapping> {
+  accessItems: AccessItem[];
+  username: string;
+  email: string | null;
+  roleType: string;
+  id: string;
+}
+
+interface SharedRoleMapping extends ASRoleMapping, WSRoleMapping {
+  accessItems: AccessItem[];
+}
+
+interface Props {
+  accessItemKey: 'groups' | 'engines';
+  singleUserRoleMappings: Array<SingleUserRoleMapping<ASRoleMapping | WSRoleMapping>>;
+  initializeSingleUserRoleMapping(roleId: string): string;
+  handleDeleteMapping(roleId: string): string;
+}
+
+const noItemsPlaceholder = <EuiTextColor color="subdued">&mdash;</EuiTextColor>;
+const invitationBadge = <EuiBadge color="hollow">{INVITATION_PENDING_LABEL}</EuiBadge>;
+
+export const UsersTable: React.FC<Props> = ({
+  accessItemKey,
+  singleUserRoleMappings,
+  initializeSingleUserRoleMapping,
+  handleDeleteMapping,
+}) => {
+  // 'accessItems' is needed because App Search has `engines` and Workplace Search has `groups`.
+  const users = ((singleUserRoleMappings as SharedUser[]).map((user) => ({
+    username: user.elasticsearchUser.username,
+    email: user.elasticsearchUser.email,
+    roleType: user.roleMapping.roleType,
+    id: user.roleMapping.id,
+    accessItems: (user.roleMapping as SharedRoleMapping)[accessItemKey],
+    invitation: user.invitation,
+  })) as unknown) as Array<Omit<SharedUser, 'elasticsearchUser | roleMapping'>>;
+
+  const columns: Array<EuiBasicTableColumn<SharedUser>> = [
+    {
+      field: 'username',
+      name: 'Username',
```

Comment:
This table column (and every other `name:` property below it) needs i18n

Reply:
https://github.com/elastic/kibana/pull/102826/commits/f156b66cd94b48debadafc817523c078404788f4